### DISCUSSION TITLE
Multitenant OPA bundle

### DISCUSSION THREAD

**Discussion by lgtti** [open-policy-agent/community#306] [OPA and Rego] [answered]
Hi, i'm trying to create an OPA configuration bundle for a multi tenant environment.
Actually my bundle structure is:
```
bundle.tar.gz
Tenant1/
Tenant1/data.json
Tenant2/
Tenant2/data.json
permissions.rego
```

permissions.rego use the 'data' variable to access json content for a specific tenant:
```
...
tenant := input.request.metadata.tenant
names := [
            name |
                regex.match(data[tenant].mappings[i].api, input.request.http.path)
                ....
        ]
...
```

A data.json is structured with a schema similar to this:
```
{
    "Tenant1":
    {
        "mappings":
      ...
```

If i try to request any policy evaluation I receive result:{allow:false} response.
If i try a single tenant without the bundle the result is result:{allow:true}, so the cause is the bundle structure or a similar problem.

What is the correct strucuture that i can use to manage one single policy file with a set of tenant data configurations?

**Comment by ashutosh-narkar** [comment, marked as answer]
Have you tried running your bundle and looking if everything looks right? You could run the below command and then run for ex. `data` in the prompt to see your data mappings. Also are you using `opa build` to create your bundle?
```
opa run bundle.tar.gz
```

**Reply by lgtti** [reply]
Thanks for you command, i have used it to verify the data variable content from the bundle.
I have solved using 
`opa test -b bundle.tar.gz -v`
addint a test rego file in the bundle to run tests using the bundle data.

The problem is the json data top level key (in my example below was "Tenant1"). Using different folders for data.json, opa loads all data in a single file in memory adding the top level key automatically.

So the result was:
```
{
    "Tenant1": {
        "Tenant1":  {
              "mappings":
      ...
```

Removing the top level ket from each data.json file solves the problem.
